src/utils/telemetryUtils.ts

```diff
@@ -10,6 +10,12 @@ export namespace TelemetryUtils {
             if (arg.subscription.subscriptionId) {
                 properties[TelemetryProperties.subscriptionId] = arg.subscription.subscriptionId;
             }
+            // data plane has no subscriptionId but subscriptionPath.
+            else if (arg.subscription.subscriptionPath && arg.subscription.tenantId && arg.subscription.userId) {
```

Comment:
Is there a case that there is subscriptionPath but not userId?

Reply:
No, once user add an api center data plane api, it should contains both `runtimeurl`, `clientid` and `tenantid` from both ui, deeplink and command palatte, these parameters will be integrated into the `subscriptionContext` by [`getSubscriptionContext`](https://github.com/microsoft/vscode-azureapicenter/blob/9ff2c475bf85ae9dd533d51294df6d583a5eae0f/src/tree/DataPlaneAccount.ts#L99)